app/Http/Controllers/Order/OrderController.php

```diff
@@ -193,14 +202,24 @@ public function show($id)
         try {
             $order = $this->order->findOrFail($id);
             $subscription = $order->subscription()->first();
-            $currenctVersion = $subscription->version;
-            $lastActivity = $subscription->updated_at;
-            $invoiceid = $order->invoice_id;
-            $invoice = $this->invoice->where('id', $invoiceid)->first();
+            $date = '--';
+            $licdate = '--';
+            $supdate = '--';
+            $connectionLabel = '--';
+            $lastActivity = '--';
+            $versionLabel = '--';
+            if($subscription) {
+              $date =  strtotime($subscription->update_ends_at)>1 ? getExpiryLabel($subscription->update_ends_at): '--';
+              $licdate = strtotime($subscription->ends_at)>1 ? getExpiryLabel($subscription->ends_at) :'--' ;
+              $supdate =  strtotime($subscription->support_ends_at)>1 ? getExpiryLabel($subscription->support_ends_at):'--' ;
+              $lastActivity = getDateHtml($subscription->updated_at).'&nbsp&nbsp;'.$this->installationStatusLabel($subscription->updated_at);
+              $versionLabel = getVersionAndLabel($subscription->version,$order->product);
+            }
+            $invoice = $this->invoice->where('id', $order->invoice_id)->first();
             if (!$invoice) {
                 return redirect()->back()->with('fails', 'no orders');
             }
-            $invoiceItems = $this->invoice_items->where('invoice_id', $invoiceid)->get();
+            $invoiceItems = $this->invoice_items->where('invoice_id', $order->invoice_id)->get();
```

Comment:
are we showing all invoices at once? no limit?

Reply:
This was not required. So removed.